jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ContentSinkOutputStream.java

```diff
@@ -96,9 +97,13 @@ public void close(Callback callback) throws IOException
     private void handleException(Throwable x) throws IOException
     {
         IOException failure = IO.rethrow(x);
-        if (failed)
-            throw new IOException(failure.toString());
-        failed = true;
+        if (failed == null)
+            failed = failure;
+        else if (ExceptionUtil.areAssociated(failed, failure))
+            failure = IO.rethrow(ExceptionUtil.copyOf(failure));
+        else
+            failed.addSuppressed(failure);
+
```

Comment:
I don't think we want to keep around a `Throwable` over a `boolean`, and I would not risk adding a large number of suppressed failures to a failure stored as a field.